microprofile/config/config/src/main/java/io/helidon/microprofile/config/MpConfig.java

```diff
@@ -73,12 +73,13 @@
         this.mpConfigSources = mpConfigSources;
 
         this.propertyNames =
-                Stream.concat(mpConfigSources.stream()
-                                      .flatMap(cs -> cs.getPropertyNames().stream()),
-                              config.traverse(Config::isLeaf)
-                                      .map(Config::key)
-                                      .map(Config.Key::toString))
-                        .collect(Collectors.toSet());
+            Stream.concat(mpConfigSources.stream()
+                              .flatMap(cs -> cs.getPropertyNames().stream()),
+                          config.traverse()
```

Comment:
I think you correctly described the intent of the original code -- harvest the leaves -- but it was overly aggressive in discarding nodes. But does the proposed change harvest _all_ keys, not just leaf ones? Is that OK?

Reply:
The original code never allowed the subsquent `map` operations to apply to anything.  The fix now makes it such that the `map` operations apply only to leaf nodes, which is what we want.

Reply:
Oh, rats, I missed the new `.filter` invocation. Sheesh.